src/Microsoft.DotNet.Interactive.Formatting/HtmlFormatter.cs

```diff
@@ -137,26 +144,36 @@ void Initialize()
         new HtmlFormatter<Type>((type, context) =>
         {
             var text = type.ToDisplayString(PlainTextFormatter.MimeType);
-                    
+
             // This is approximate
             var isKnownDocType =
                 type.Namespace is not null &&
+                type.FullName is not null &&
                 (type.Namespace == "System" ||
                  type.Namespace.StartsWith("System.") ||
-                 type.Namespace.StartsWith("Microsoft."));
+                 type.Namespace.StartsWith("Microsoft.")) &&
+                !type.IsAnonymous();
```

Comment:
Consider moving the check for anonymous types earlier since its probably less expensive than the string comparisons above.

Reply:
We own the `IsAnonymous` method and it's likely the most expensive of the operations in this expression.

Reply:
Ah interesting - my bad I assumed it was a BCL API that would be checking some flag in the metadata.